File: apk/app/src/main/java/com/example/momentra/analytics/TrackScreen.kt
package com.example.momentra.analytics

import androidx.compose.runtime.Composable
import androidx.compose.runtime.DisposableEffect

@Composable
fun TrackScreen(
    screenName: String,
    screenClass: String = screenName,
    content: @Composable () -> Unit = {},
) {
    DisposableEffect(screenName) {
        MomentraAnalytics.get().onScreenEnter(screenName, screenClass)
        onDispose { MomentraAnalytics.get().onScreenExit(screenName) }
    }
    content()
}

fun trackWidget(screenName: String, widgetName: String, action: String = "tap") {
    MomentraAnalytics.get().trackWidget(screenName, widgetName, action)
}

File: momentra/momentra/Analytics/TrackScreen.swift
import SwiftUI

private struct TrackScreenModifier: ViewModifier {
    let screenName: String
    let screenClass: String

    func body(content: Content) -> some View {
        content
            .onAppear {
                MomentraAnalytics.shared.onScreenEnter(screenName, screenClass: screenClass)
            }
            .onDisappear {
                MomentraAnalytics.shared.onScreenExit(screenName)
            }
    }
}

extension View {
    func trackScreen(_ screenName: String, screenClass: String? = nil) -> some View {
        modifier(TrackScreenModifier(
            screenName: screenName,
            screenClass: screenClass ?? screenName
        ))
    }
}

func trackWidget(screenName: String, widgetName: String, action: String = "tap") {
    MomentraAnalytics.shared.trackWidget(screenName: screenName, widgetName: widgetName, action: action)
}
